### PR TITLE
Test for PR #9486, Validate allowed TLDs with current IANA list

### PR DESCRIPTION
**Pull request by sebastianpiskorski**
fixes #9486

### CODE REVIEW

File: tests/PHPUnit/Integration/EmailValidatorTest.php [hunk 1]
```diff
@@ -8,6 +8,9 @@
 
 namespace Piwik\Tests\Integration;
 
+use Guzzle\Http\Client;
+use Guzzle\Http\Message\Request;
+use Guzzle\Http\Message\Response;
```
Comment (tsteur) [inline tests/PHPUnit/Integration/EmailValidatorTest.php]: FYI: Will guzzle is a library in Piwik it is only their for AWS-SDK. It's not a supported lib and we might remove it at some point. Instead use `Piwik\Http`.

File: tests/PHPUnit/Integration/EmailValidatorTest.php [hunk 2]
```diff
@@ -20,6 +23,40 @@ protected function isValid($email)
...
+            $this->markTestSkipped("Couldn't get TLD list");
+        }
+
+        foreach ($tlds as $key => $tld) {
+            if (strpos(mb_strtolower($tld), 'xn--') !== 0) {
+                $tld = mb_strtolower($tld);
+            }
+            $this->assertTrue(
+                $this->isValid('test@example.' . idn_to_utf8($tld))
+            );
```
Comment (tsteur) [inline tests/PHPUnit/Integration/EmailValidatorTest.php]: Can we also add some negative cases? I presume so far all the tests would make work even if the method only always returned true :) (apart from one other test)

**Comment by sebastianpiskorski** [comment]
I will fix those soon.

File: tests/PHPUnit/Integration/EmailValidatorTest.php (hunk 2 above)
Comment (sebastianpiskorski) [inline tests/PHPUnit/Integration/EmailValidatorTest.php]: done

**Comment by sebastianpiskorski** [comment]
Introduced CR suggestions.

**Comment by mattab** [comment]
@sebastianpiskorski @tsteur This PR introduces a little challenge, see:  Automated test for allowed TLDs starts failing when iana.org starts supporting new domain extensions #9977

**Comment by tsteur** [comment]
I think it's supposed to fail in such a case to make sure it'll work for all domains in the future. That's why I merged it. Not sure how often they add new domains though. If they add new ones too often could maybe solve it differently

**Pull request merged**